test/suites/drift/suite_test.go

```diff
@@ -148,20 +149,89 @@ var _ = Describe("Drift", Label("AWS"), func() {
+		Expect(err).To(BeNil())
+
+		By("creating a new securitygroup")
+		securitygroup := env.GetSecurityGroups(map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName})
+		Expect(len(securitygroup)).To(BeNumerically("==", 2))
+
+		createSecurityGroup := &ec2.CreateSecurityGroupInput{
+			GroupName:   awssdk.String("security-group-drift"),
+			Description: awssdk.String("End-to-end Drift Test, should delete after drift test is completed"),
+			VpcId:       output.Cluster.ResourcesVpcConfig.VpcId,
+			TagSpecifications: []*ec2.TagSpecification{
+				{
+					ResourceType: awssdk.String("security-group"),
+					Tags: []*ec2.Tag{
+						{
+							Key:   awssdk.String("security-group-drift"),
+							Value: awssdk.String(settings.FromContext(env.Context).ClusterName),
+						},
+					},
+				},
+			},
+		}
+		newSecurityGroup, err := env.EC2API.CreateSecurityGroup(createSecurityGroup)
+		Expect(err).To(BeNil())
+		DeferCleanup(func() {
+			By("deleting the new securitygroup")
+			// // Need to make sure that all instance launched with the security group are terminated
+			EventuallyWithOffset(1, func(g Gomega) {
+				output, err := env.EC2API.DescribeNetworkInterfaces(&ec2.DescribeNetworkInterfacesInput{
+					Filters: []*ec2.Filter{{Name: awssdk.String("group-id"), Values: []*string{newSecurityGroup.GroupId}}},
+				})
+				g.Expect(err).To(BeNil())
+				_, err = env.EC2API.ModifyNetworkInterfaceAttribute(&ec2.ModifyNetworkInterfaceAttributeInput{
+					NetworkInterfaceId: output.NetworkInterfaces[0].NetworkInterfaceId,
+					Groups:             lo.Map(securitygroup, func(sg aws.SecurityGroup, _ int) *string { return sg.GroupId }),
+				})
+				g.Expect(err).To(BeNil())
+				_, err = env.EC2API.DeleteSecurityGroup(&ec2.DeleteSecurityGroupInput{
+					GroupId: newSecurityGroup.GroupId,
+				})
+				g.Expect(err).To(BeNil())
+			}).Should(Succeed())
+		})
+		By("creating a new provider with the new securitygroup")
+		provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+			AWS: v1alpha1.AWS{
+				SecurityGroupSelector: map[string]string{"aws-ids": fmt.Sprintf("%s,%s,%s", *securitygroup[0].GroupId, *securitygroup[1].GroupId, awssdk.StringValue(newSecurityGroup.GroupId))},
+				SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			},
+		})
+		provisioner := test.Provisioner(test.ProvisionerOptions{ProviderRef: &v1alpha5.MachineTemplateRef{Name: provider.Name}})
+
+		// Add a do-not-evict pod so that we can check node metadata before we deprovision
+		pod := test.Pod(test.PodOptions{
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
+					v1alpha5.DoNotEvictPodAnnotationKey: "true",
+				},
+			},
+		})
+
+		env.ExpectCreated(pod, provider, provisioner)
+		env.EventuallyExpectHealthy(pod)
+		env.ExpectCreatedNodeCount("==", 1)
+		By("updating the provider securitygroup")
+		node := env.Monitor.CreatedNodes()[0]
```

Comment:
I think you can use the following to be more deterministic about which node we're talking about.
`node := env.ExpectScheduled(pod)`

We don't want any other accidentally created nodes messing with this test.

Reply:
Isn't `env.ExpectScheduled(pod)` only for our unit testing?

Reply:
You're absolutely right 🤦 , never mind!